### PR TITLE
[Fixes #40] Adjust stuff for CI / pre-release

### LINKED ISSUE #40: CI is Broken
The CI is broken since the last push.
The CI for #38 fails in a few places (where it decides to run).
First cause of failure is that the CI think every subfolder of `./pacs/` is a pac, it clears the two first characters `at` normally, and builds with the selected pac as a feature.
Now there is a `templates` folder resulting in an attempt to run `cargo build --features unproven,mplates`.
Second Cause of failure mostly speaks for itself:
```
`cargo metadata` exited with an error: error: current package believes it's in a workspace when it's not:
current:   /home/runner/work/atsamx7x-rust/atsamx7x-rust/boards/atsame70_xpro/Cargo.toml
workspace: /home/runner/work/atsamx7x-rust/atsamx7x-rust/Cargo.toml

this may be fixable by adding `boards/atsame70_xpro` to the `workspace.members` array of the manifest located at: /home/runner/work/atsamx7x-rust/atsamx7x-rust/Cargo.toml
Alternatively, to keep it out of the workspace, add the package to the `workspace.exclude` array, or add an empty `[workspace]` table to the package's manifest.
cargo-fmt 
```
and results from the changes made to `Cargo.toml` in the root folder.

### PR DESCRIPTION
**Pull request by michalfita**
Preparing stuff for HAL 0.4.0 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
 ### Added
 
 - GitHub CI.
+- Templates for PACs restored, so we create both correct `Cargo.toml` and `README.md` required to publish to `crates.io`.
 
 ### Changed
 
 - [Sealed](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) all HAL-internal traits to disallow potentially unsound downstream trait implementations.
 - Regenerated PACs with `svd2rust 0.25.1` using up-to-date (2022-05-18) SVDs. These SVDs are now vendored in this repository, instead of using a submodule.
+- Cargo Workspace sorted, so local builds and crate publictions work, and CI is satisfied.
```
Comment (tmplt) [inline CHANGELOG.md]: GitHub CI was added with in this release, so there is no need to mention that a fix was applied.

**Review by tmplt — CHANGES_REQUESTED** [review]

**Review by martinmortsell — APPROVED** [review]
Looks good to me.

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,7 +13,8 @@
 
 - [Sealed](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) all HAL-internal traits to disallow potentially unsound downstream trait implementations.
 - Regenerated PACs with `svd2rust 0.25.1` using up-to-date (2022-05-18) SVDs. These SVDs are now vendored in this repository, instead of using a submodule.
-- Cargo Workspace sorted, so local builds and crate publictions work, and CI is satisfied.
+- Cargo Workspace sorted, so local builds and crate publictions work.
```
Comment (tmplt) [inline CHANGELOG.md]: spelling

File: tools/update-pacs.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    # Find chipname, and create an empty PAC directory
+    chip=$(basename "${svd}" .svd)
+    crate=$(echo ${chip} | tr '[:upper:]' '[:lower:]')
+    pac="${top}/pac/${crate}"
+    rm -rf $pac
+    mkdir -p $pac
+    pushd $pac
+
+    #sed -e "s|\${crate}|${crate}|g" -e "s|\${chip}|${chip}|g" -e "s|\${pac}|${pac}|g" -e "s|\${version}|${version}|g" "${top}/pac/templates/Cargo.toml.template" > Cargo.toml
+    #sed -e "s|\${crate}|${crate}|g" -e "s|\${chip}|${chip}|g" -e "s|\${pac}|${pac}|g" -e "s|\${version}|${version}|g" "${top}/pac/templates/README.md.template" > README.md
```
Comment (tmplt) [inline tools/update-pacs.sh]: remove outcommented code

**Review by tmplt — CHANGES_REQUESTED** [review]

**Review by tmplt — APPROVED** [review]

**Comment by michalfita** [comment]
[0.4.0 Released](https://github.com/atsams-rs/atsamx7x-rust/releases/tag/v0.4.0)!

**Pull request merged**